csds-client/main.go

```diff
@@ -2,11 +2,64 @@ package main
 
 import (
 	"envoy-tools/csds-client/client"
+	client_v2 "envoy-tools/csds-client/client/v2"
+	"flag"
 	"log"
+	"time"
 )
 
+// GetClientOptionsAndApiVersionFromFlags parses flags to ClientOptions and api version
+func GetClientOptionsAndApiVersionFromFlags() (client.ClientOptions, string) {
+	const uriDefault string = "trafficdirector.googleapis.com:443"
+	const platformDefault string = "gcp"
+	const authnModeDefault string = "auto"
+	const apiVersionDefault string = "v2"
+	const requestFileDefault string = ""
+	const requestYamlDefault string = ""
+	const jwtDefault string = ""
+	const configFileDefault string = ""
+	const monitorIntervalDefault time.Duration = 0
+	const visualizationDefault bool = false
+
+	uriPtr := flag.String("service_uri", uriDefault, "the uri of the service to connect to")
```

Comment:
You don't really need this function all together. Typically initialization of flags and what you do with the values should be decoupled. If you did it, you wouldn't need to create function with overloaded signature like GetXandYandX()...

Instead, I suggest following:
1. Create and initialize flags in the beginning of main code, see some examples here: https://golang.org/pkg/flag/
2. Build client option inline inside main using flag values. You can keep it in the function as well, but it seems to add little to no value at that point.

Reply:
Modified this to

1. Create flag variables as global variables

2. Bind flag with variables in `init()`

3. Parse flags and build ClientOptions inline